File: src/backend/DonkeyWork.Chat.Api/Models/Prompt/GetActionPromptsModel.cs
// ------------------------------------------------------
// <copyright file="GetActionPromptsModel.cs" company="DonkeyWork.Dev">
// Provided as is by (c) DonkeyWork.Dev. No warranty or support is given.
// </copyright>
// ------------------------------------------------------

namespace DonkeyWork.Chat.Api.Models.Prompt;

/// <summary>
/// Gets the action prompts model.
/// </summary>
public record GetActionPromptsModel
{
    /// <summary>
    /// Gets the count of action prompts.
    /// </summary>
    public int Count { get; init; }

    /// <summary>
    /// Gets the action prompts.
    /// </summary>
    public IEnumerable<GetActionPromptsItemModel> Prompts { get; init; } = [];
}

File: src/ui/src/models/api/Prompt/GetActionPromptsModel.ts
import { ActionPromptItem } from './GetActionPromptsItemModel';

/**
 * Model for the paginated list of action prompts returned from the API
 */
export interface GetActionPromptsModel {
  /**
   * The total number of action prompts
   */
  total: number;
  
  /**
   * The count of action prompts
   */
  count?: number;
  
  /**
   * The list of action prompts
   */
  items?: ActionPromptItem[];
  
  /**
   * Backward compatibility field
   */
  prompts?: ActionPromptItem[];
}